ping/ping_common.c

```diff
@@ -941,25 +949,48 @@ int finish(struct ping_rts *rts)
 
 		tmdev = llsqrt(tmvar);
 
-		printf(_("rtt min/avg/max/mdev = %ld.%03ld/%lu.%03ld/%ld.%03ld/%ld.%03ld ms"),
-		       (long)rts->tmin / 1000, (long)rts->tmin % 1000,
-		       (unsigned long)(tmavg / 1000), (long)(tmavg % 1000),
-		       (long)rts->tmax / 1000, (long)rts->tmax % 1000,
-		       (long)tmdev / 1000, (long)tmdev % 1000);
+		char rttmin[30];
+		char rttavg[30];
+		char rttmax[30]; 
+		char rttmdev[30];
+
+		sprintf(rttmin, "%ld.%03ld", (long)rts->tmin / 1000, (long)rts->tmin % 1000);
+		sprintf(rttavg, "%lu.%03ld", (unsigned long)(tmavg / 1000), (long)(tmavg % 1000));
+		sprintf(rttmax, "%ld.%03ld", (long)rts->tmax / 1000, (long)rts->tmax % 1000);
+		sprintf(rttmdev, "%ld.%03ld", (long)tmdev / 1000, (long)tmdev % 1000);
+
+		if (!rts->opt_json)
+			printf(_("rtt min/avg/max/mdev = %s/%s/%s/%s ms"),
+				rttmin, rttavg, rttmax, rttmdev);
+
+		construct_json_statistics(rts, tv, rttmin, rttavg, rttmax, rttmdev);
```

Comment:
Th code looks much better, but why don't we add a function `ping_print_statistics()` into the `ping_output.h` and put all this code in there? And the same tome for the flood option.

If I was writing this patch, most if not all the `if (!rts->opt_json)` would have been in the `ping_output.c`.

Reply:
Sure that sounds good, and I refactored it further now to keep all the JSON conditions in `ping_output.c` as well as to deduplicate some of the printing logic between ping and ping6, but my main issue still is that I don't know how to test some of the code branches I am touching, so with such big printing refactoring there is a good chance I might break formatting of some odd error conditions.